draft-ietf-quic-version-negotiation.md

```diff
@@ -194,92 +215,49 @@ server.
 During the handshake, endpoints will exchange handshake version information,
 which is a blob of data that is defined below. In QUIC version 1, the handshake
 version information is transmitted using a new transport parameter,
-`version_negotiation`. The contents of handshake version information depend on
-whether the client or server is sending it, and are shown below (using the
-notation from the "Notational Conventions" section of {{QUIC}}):
+`version_negotiation`. The contents of handshake version information are shown
+below (using the notation from the "Notational Conventions" section of
+{{QUIC}}):
 
 ~~~
-Client Handshake Version Information {
+Handshake Version Information {
   Currently Attempted Version (32),
-  Previously Attempted Version (32),
-  Received Negotiation Version Count (i),
-  Received Negotiation Version (32) ...,
-  Compatible Version Count (i),
-  Compatible Version (32) ...,
+  Other Versions (32) ...,
 }
 ~~~
-{: #fig-client-hvi title="Client Handshake Version Information"}
+{: #fig-hvi-format title="Handshake Version Information Format"}
 
 The content of each field is described below:
 
 Currently Attempted Version:
 
-: The version that the client is attempting to use. This field MUST be equal to
-the value of the Version field in the long header that carries this data.
+: This field SHALL be equal to the value of the `Version` field in the long
+header that carries this data.
 
-Previously Attempted Version:
+The contents of the `Other Versions` field depends on whether it is sent by the
+client or by the server.
 
-: If the client is sending this first flight in response to a Version
-Negotiation packet, this field contains the version that the client used in the
-previous first flight that triggered the version negotiation packet. If the
-client did not receive a Version Negotiation packet, this field SHALL be
-all-zeroes.
+Client-Sent Other Versions:
 
-Received Negotiation Version Count:
+: When sent by a client, the `Other Versions` field lists all the versions that
+this first flight is compatible with, ordered by descending preference. Note
+that the version in the `Currently Attempted Version` field MUST be included in
+this list to allow the client to communicate the currently attempted version's
+preference. Note that this preference is only advisory, servers MAY choose to
+use their own preference instead.
```

Comment:
I'm still not on board with preference ordering.  But let's track that separately.

Reply:
I filed #43 to track this.